### PR TITLE
[895, 900, 905] Weight

### PR DESCRIPTION
**Pull request by moderatemisbehaviour**
* Extracted weight from *nh_observations* into *nh_weight*.
* Added the *waist_measurement* field.
* Added BMI calculation stored in the *score* field.

### CODE REVIEW

File: nh_eobs_api/__init__.py [hunk 1]
```diff
@@ -1,3 +1,8 @@
 # Part of Open eObs. See LICENSE file for full copyright and licensing details.
 from . import routing
 from . import controllers
+
+# Fix JSON encoding of floats.
+from json import encoder
+encoder.c_make_encoder = None
+encoder.FLOAT_REPR = lambda o: format(o, '.1f')
```
Comment (moderatemisbehaviour) [inline nh_eobs_api/__init__.py]: Add tech debt.

File: nh_eobs_api/controllers/route_api.py [hunk 2]
```diff
@@ -350,29 +356,18 @@ def process_ajax_form(self, *args, **kw):
...
-        rel_tasks = ''
-        if len(triggered_tasks):
-            rel_tasks = 'Here are related tasks based on the observation'
+        activity = activity_api.browse(cr, uid, int(task_id))
+        obs = activity.data_ref
+
+        description = self.get_submission_message(obs)
+        response_data = obs.get_submission_response_data()
+
         response_json = ResponseJSON.get_json_data(
```
Comment (moderatemisbehaviour) [inline nh_eobs_api/controllers/route_api.py]: Swap all `test` dictionaries for `_logger.debug`.

File: nh_ews/tests/nh_clinical_patient_observation_ews/test_get_submission_message.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from openerp.tests.common import TransactionCase
+
+
+class TestGetSubmissionMessage(TransactionCase):
+
+    def setUp(self):
+        super(TestGetSubmissionMessage, self).setUp()
```
Comment (moderatemisbehaviour) [inline nh_ews/tests/nh_clinical_patient_observation_ews/test_get_submission_message.py]: Add unit tests.

File: nh_observations/tests/nh_clinical_patient_observation/test_get_triggered_tasks.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        pass
+
+    def test_triggered_tasks_returns_list_of_records(self):
+        pass
+
+    def test_triggered_obs_not_included(self):
+        pass
+
+    def test_no_creator_id_raises_exception(self):
+        pass
```
Comment (moderatemisbehaviour) [inline nh_observations/tests/nh_clinical_patient_observation/test_get_triggered_tasks.py]: Add unit tests.

File: nh_weight/models/nh_clinical_patient_observation_weight.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+    @api.model
+    def get_form_description(self, patient_id):
+        """
+        Override of `nh.clinical.patient.observation_scored` method.
+
+        :param patient_id:
+        :return:
+        """
+        form_description = copy.deepcopy(self._form_description)
```
Comment (moderatemisbehaviour) [inline nh_weight/models/nh_clinical_patient_observation_weight.py]: Look into if this is necessary.

File: nh_weight/tests/nh_clinical_patient_observation_weight/test_get_data_visualisation_resource.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from openerp.tests.common import TransactionCase
+
+
+class TestGetDataVisualisationResource(TransactionCase):
+
+    def setUp(self):
+        super(TestGetDataVisualisationResource, self).setUp()
```
Comment (moderatemisbehaviour) [inline nh_weight/tests/nh_clinical_patient_observation_weight/test_get_data_visualisation_resource.py]: Add a test

File: nh_weight/sql/wardboard.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# from openerp import models
+
+
+# class WardboardSql(models.AbstractModel):
+
+# _inherit = 'nh.clinical.sql'
+# _name = 'nh.clinical.sql'
+
+# Add this to the SELECT of the wardboard_skeleton
```
Comment (moderatemisbehaviour) [inline nh_weight/sql/wardboard.py]: Move to nh_monitoring

**Comment by Gimpneek** [comment]
:shipit: Once Travis turns green (due to outage their end)

**Comment by moderatemisbehaviour** [comment]
Build is failing and it may be due to problems with Travis according to [this status update](https://www.traviscistatus.com/incidents/j7f0sp50n9xw).
Open & closing PR as advised.

**Comment by Gimpneek** [comment]
new fixes LGTM :shipit:

**Comment by moderatemisbehaviour** [comment]
LGMT :shipit: :+1:

**Comment by Nayira** [comment]
Code review LGTM :shipit:

**Comment by Nayira** [comment]
Code review LGTM :shipit:

**Comment by Nayira** [comment]
Test: Pass :shipit: 
Stories: EOBS-900, EOBS-905 and EOBS-895
Issues: EOBS-1056, EOBS-1068 (both from F&F), EOBS-1098, EOBS-1099, EOBS-1100, EOBS-1101, EOBS-1102, EOBS-1103, EOBS-1104, EOBS-1105, EOBS-1107, EOBS-1126, EOBS-1092 (list of obs)

**Pull request merged**